paddle/framework/op_desc.proto

```diff
@@ -51,6 +51,9 @@ message OpDesc {
     // type of this Operator, such as "add", "sub", "fc".
     required string type = 3;
 
```

Comment:
We have a offline discussion about this.
There are commonly two steps for users training a neural network:
- users describe the construction of neural network
- users set a certain device, and run the network on certain device.   

Since we decide to expose operators to users and allow users to use operators to construction our network, operator should not have device information. 

After the network is defined and before the networks runs, we will set a certain device. And an operator's  running method can be various on different devices. The implementation of an operator's running method is defined as kernel.

After device is set, we'd better to decide which kernels will be launched before the network runs.(Or, we can decide on the fly)